src/components/RightPane/CoursePane/SearchForm/TermSelector.tsx

```diff
@@ -32,6 +46,24 @@ class TermSelector extends PureComponent<TermSelectorProps> {
     handleChange = (event: ChangeEvent<{ name?: string | undefined; value: unknown }>) => {
         this.setState({ term: event.target.value });
         this.props.changeState(this.props.fieldName, event.target.value as string);
+    
+        let stateObj = { url: "url" };
+        const url = new URL(window.location.href)
+        const urlParam = new URLSearchParams(url.search);
+        urlParam.delete('term');
+        if (event.target.value as string != "" && event.target.value as string != null){
+            urlParam.append('term', event.target.value as string);
+            const new_url = `?${urlParam.toString()}`;
+            history.replaceState(stateObj, "url", "/" + new_url);
+        }else{
+            if (urlParam.toString() == "" || urlParam.toString() == null){
+                const new_url = `${urlParam.toString()}`;
+                history.replaceState(stateObj, "url", "/" + new_url);
+            }else{
+                const new_url = `?${urlParam.toString()}`;
+                history.replaceState(stateObj, "url", "/" + new_url);
+            }
+        }
```

Comment:
Both of the bottom clauses look the same, and they all terminate similarly? So I think this might be simpler.
```suggestion
        if (typeof event.target.value === 'string' && event.target.value) {
            urlParam.append('term', event.target.value);
        } 
        const new_url = `${urlParam.toString()}`;
        history.replaceState(stateObj, "url", "/" + new_url);
```

Reply:
In fact there is a little bit difference here. The the main problem is if this changing value is the first change in the url, then it has to stringed with`?` in url  to let values after it work as a parameter. And if the changing value isn't the first one, then it should not append more `?` into the url.

Reply:
Oh oops I missed that, what about something like this? It looks kinda funny, but I think it's better than making an control branch to handle this simple case?
```ts
   const param = urlParam.toString()
   const new_url = `${param && param !== 'null' ? '?' : ''}${param}`;
   history.replaceState(stateObj, "url", "/" + new_url);
```